docs/ApplicationRatings.md

```diff
@@ -0,0 +1,82 @@
+named `Rating (stars)`.  `Rating (stars)` is a field type that displays 5 stars and fills in the stars
+ based on a calculated value.  This field type is a slightly modified version of the one
+provided by sugarcrmlabs:  [RatingStarField](https://github.com/sugarcrmlabs/RatingStarField).
+
+We created a 
+[custom field](http://support.sugarcrm.com/Documentation/Sugar_Versions/7.9/Ent/Administration_Guide/Developer_Tools/Studio/#Creating_Fields)
+named `programmingscore_c` in the Applicants (Leads) module.  The `programmingscore_c` field is not displayed in any
+views.  The job of the `programmingscore_c` field is to hold the applicant's score for the programming languages
+they know.  Because we can't use Sugar Logic with MultiSelect fields, we use a 
+[before_save logic hook](http://support.sugarcrm.com/Documentation/Sugar_Developer/Sugar_Developer_Guide_8.1/Architecture/Logic_Hooks/Module_Hooks/before_save/index.html)
+to update this field whenever an update is made to the `programminglanguages_c` field.  The files associated with the
+logic hook are 
+[update_applicant_programming_language_score.php](../package/src/custom/Extension/modules/Leads/Ext/LogicHooks/update_applicant_programming_language_score.php)
+and 
+[ApplicantProgrammingScore.php](../package/src/custom/modules/Leads/ApplicantProgrammingScore.php).
+
+We created a 
+[custom field](http://support.sugarcrm.com/Documentation/Sugar_Versions/7.9/Ent/Administration_Guide/Developer_Tools/Studio/#Creating_Fields)
+named `rating_c` in the Applicants (Leads) module.  The `rating_c` field is of type `Rating (stars)`.  The 
+`rating_c` field uses 
+[Sugar Logic](http://support.sugarcrm.com/Knowledge_Base/Studio_and_Module_Builder/Sugar_Logic/index.html) to calculate 
+the application's rating.  The formula it uses to sum the points for the applicant's GPA and programming score is 
+`add(multiply($gpa_c,10),$programming_score_c)`.
+
+We also updated the existing `gpa_c` field to have 
+[custom field validation](http://support.sugarcrm.com/Documentation/Sugar_Developer/Sugar_Developer_Guide_8.0/Cookbook/Adding_Field_Validation_to_the_Record_View/index.html) 
+so the value must be at least 0 but not higher than 4.0.  The files associated with the custom field validation are
+[create.js](../package/src/custom/modules/Leads/clients/base/views/create/create.js), 
+[record.js](../package/src/custom/modules/Leads/clients/base/views/record/record.js), and
+[en_us.error_custom_message.php](../package/src/custom/Extension/application/Ext/Language/en_us.error_custom_message.php).
+
+## Extensions
+
+| Module | Extension | Name | Description |
+| --- | --- | ---| ---|
+| Leads | Vardefs | `rating_c` | "Rating (stars)" field that displays the applicant's rating. |
+| Leads | Vardefs | `programmingscore_c` | Float field that holds the applicant's programming score. This field is not displayed. |
```

Comment:
You also used a Language extension to add error message. You also used Logic Hooks extension to register the hook to Leads module.